src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -240,264 +212,345 @@ InferenceEngine::Precision Snippet::getRuntimePrecision() const {
+    // Here it must be all the stuff that could be done once for both static and dynamic shapes
+
+    masterShape = getNormalizedDimsBySize(masterShape, tensorRank);
+    for (auto& pshape : normInputShapes)
+        pshape = getNormalizedDimsBySize(pshape, tensorRank);
+    for (auto& pshape : normOutputShapes)
+        pshape = getNormalizedDimsBySize(pshape, tensorRank);
+
+    tileRank = 1;
+    fullWorkAmount = std::accumulate(masterShape.begin(), masterShape.end(), 1, std::multiplies<size_t>());
+    // optimizeExecDomain will collapse shape dimensions and adjust tile Rank
+    optimizeExecDomain(normInputShapes, normOutputShapes, masterShape, tileRank);
+    exec_domain = masterShape;
+
+    // todo: probably better to pass a call_args instance
+    calcJITParams(data_offsets);
+    auto initStartMemoryOffsets = [this]() {
+        const auto config = getSelectedPrimitiveDescriptor()->getConfig();
+        const size_t numInputs = inputShapes.size();
+        start_offset_in.resize(numInputs);
+        srcMemPtrs.resize(numInputs);
+        for (size_t i = 0; i < numInputs; i++) {
+            const auto memPtr = getParentEdgeAt(i)->getMemoryPtr();
+            srcMemPtrs[i] = memPtr;
+            start_offset_in[i] =  memPtr->GetDescWithType<BlockedMemoryDesc>()->getOffsetPadding() * dataSize[i];
+        }
+        const size_t numOutputs = outputShapes.size();
+        start_offset_out.resize(numOutputs);
+        dstMemPtrs.resize(numOutputs);
+        for (size_t i = 0; i < numOutputs; i++) {
+            const auto memPtr = getChildEdgeAt(i)->getMemoryPtr();
+            dstMemPtrs[i] = memPtr;
+            start_offset_out[i] = memPtr->GetDescWithType<BlockedMemoryDesc>()->getOffsetPadding() * dataSize[i + numInputs];
+        }
+    };
+    // initialize start offsets to src and dst memory
+    // Needs to be done for every set of input shapes sce memory ptrs could've updated
+    initStartMemoryOffsets();
+    std::vector<size_t> scheduler_work_amounts;
+    // rename schedulerWorkAmount to harnessWorkAmount?
+    harnessWorkAmount = fullWorkAmount;
+    const auto rank = exec_domain.size();
+    for (auto i = rank - tileRank; i < rank; i++) {
+        auto& dim = exec_domain[i];
+        harnessWorkAmount /= dim;
+        scheduler_work_amounts.push_back(dim);
+        dim = 1;
+    }
 
-    initOffsets();
-    initSchedulingInfo();
+    std::vector<ov::Shape> new_shapes;
+    for (const auto& s : normInputShapes) {
+        ov::Shape ns(tileRank, 0);
+        const int offset = s.size() - tileRank;
+        // todo: this check is excessive, remove it before merge
+        if (offset < 0)
+            IE_THROW() << "Error during creating reduced body shapes: tileRank is larger than the input size";
+        std::copy(s.begin() + offset, s.end(), ns.begin());
+        new_shapes.emplace_back(std::move(ns));
+    }
+    // Before body reshaping we should scale axis of Softmax
+    auto ops = snippet->get_body()->get_ops();
```

Comment:
We agreed not to perform reshaping and to pass plugin shapes in rt_info. Please refer to the transpose PR.